### PR TITLE
Drop Python 3.4 support

### PR DESCRIPTION
**Pull request by laixintao**
## Description

https://www.python.org/downloads/release/python-340/

related: #1139 

## Checklist

- [x] I've added this contribution to the `changelog.rst`.
- [x] I've added my name to the `AUTHORS` file (or it's already there).

- [x] I installed pre-commit hooks (`pip install pre-commit && pre-commit install`), and ran `black` on my code.
- [x] Please squash merge this pull request (uncheck if you'd like us to merge as multiple commits)

### CODE REVIEW

File: changelog.rst [hunk 1]
```diff
@@ -16,6 +16,11 @@ Internal:
 
 * Fix dead link in development guide. (Thanks: `BrownShibaDog`_)
 
+Deprecation:
```
Comment (j-bennet) [inline changelog.rst]: Let's call this section "Internal".

File: changelog.rst [hunk 2]
```diff
@@ -16,6 +16,11 @@ Internal:
...
+------------
+
+* Drop Python3.4 support.
```
Comment (j-bennet) [inline changelog.rst]: Please add your name here - see other changelog entries for an example.

**Comment by laixintao** [comment]
> Nice work! Please address the changelog comments, and remove 3.4 from tox.ini as well.

Oh sorry, I search `3.4` for the whole repo but missed here :)

**Comment by j-bennet** [comment]
Merged! Thanks a lot!

**Pull request merged**